src/material-experimental/mdc-chips/chip-cell.ts

```diff
@@ -6,14 +6,271 @@
 
+let uid = 0;
+
+/** Represents an event fired on an individual `mat-chip`. */
+export interface MatChipCellEvent {
+  /** The chip the event was fired on. */
+  chip: MatChipCell;
+}
+
+/**
+ * Directive to add CSS classes to chip leading icon.
+ * @docs-private
+ */
 @Directive({
-  selector: 'mat-chip-cell',
+  selector: 'mat-chip-avatar, [matChipAvatar]',
+  host: {
+    'class': 'mat-chip-avatar mdc-chip__icon mdc-chip__icon--leading',
+  }
+})
+export class MatChipAvatar {
+  constructor(private _changeDetectorRef: ChangeDetectorRef,
+              private _elementRef: ElementRef) {}
+
+  /** Sets whether the given CSS class should be applied to the leading icon. */
+  setClass(cssClass: string, active: boolean) {
+    const element = this._elementRef.nativeElement;
+    active ? element.addClass(cssClass) : element.removeClass(cssClass);
+    this._changeDetectorRef.markForCheck();
+  }
+}
+
+/**
+ * Directive to add CSS class to chip trailing icon and notify parent chip
+ * about trailing icon interactions.
+ *
+ * If matChipRemove is used to add this directive, the parent chip will be
+ * removed when the trailing icon is clicked.
+ *
+ * @docs-private
+z */
+@Directive({
+  selector: 'mat-chip-trailing-icon, [matChipTrailingIcon], [matChipRemove]',
+  host: {
+    'class': 'mat-chip-trailing-icon mdc-chip__icon mdc-chip__icon--trailing',
+    'tabindex': '0',
+    'role': 'button',
+    '(click)': 'interaction.emit($event)',
+    '(keydown)': 'interaction.emit($event)',
+  }
+})
+export class MatChipTrailingIcon {
+  @Output() interaction = new EventEmitter<MouseEvent | KeyboardEvent>();
+  shouldRemove!: boolean;
+
+  constructor(_elementRef: ElementRef) {
+    this.shouldRemove = _elementRef.nativeElement.getAttribute('matChipRemove') !== null;
+  }
+}
+
+/**
+ * Material design styled Chip component. Used inside the MatChipGrid component.
+ */
+@Component({
+  selector: 'mat-chip-cell, mat-basic-chip-cell',
+  templateUrl: 'chip-cell.html',
+  styleUrls: ['chips.css'],
   host: {
     'class': 'mat-mdc-chip-cell',
+    '[id]': 'id',
+    '[class.mat-chip-disabled]': 'disabled',
+    '[class.mat-chip-selected]': 'selected',
   },
 })
-export class MatChipCell {
-  // TODO: set up MDC foundation class.
+export class MatChipCell implements AfterContentInit, AfterViewInit, OnDestroy {
+  private _uniqueId = `mat-mdc-chip-${uid++}`;
+
+  /** A unique id for the chip. If none is supplied, it will be auto-generated. */
+  @Input() id: string = this._uniqueId;
+
+  /** The color palette for this chip ('primary', 'accent', or 'warn'). */
+  @Input() color: ThemePalette;
+
+  private _disabled = false;
+
+  /** Whether the chip is disabled. */
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+  set disabled(disabled) {
+    this._disabled = coerceBooleanProperty(disabled);
+    this._changeDetectorRef.markForCheck();
+  }
+
+  /** Whether the chip is selected. */
+  @Input()
+  get selected(): boolean {
```

Comment:
I was thinking that we probably want to get rid of the notion of "selected" for at least grid-based chips, since that doesn't semantically mean anything for grid cells (though we would still have to keep it for the listbox-based model). 

We could probably support something more generic, like "highlight" at the base level, and then use that to display the selection state for the variants that do support it.